dask_distance/_utils.py

```diff
@@ -70,6 +70,20 @@ def _cdist_apply(U, V, metric):
     return result
 
 
+def _pdist_apply(UI, VJ, U, V, metric):
+    U = U[:, 0, :]
+    V = V[0, :, :]
+
+    result = numpy.empty((len(U), len(V)), dtype=float)
+
+    for (i,), ui in numpy.ndenumerate(UI.flat):
+        for (j,), vj in numpy.ndenumerate(VJ.flat):
+            if ui < vj:
+                result[i, j] = metric(U[i], V[j])
```

Comment:
Tried playing with various NumPy tricks to get rid of this `for`-loop. However it seems that this still ends up being the fastest way to do this check.

Reply:
Appears this is mainly due to the fact that doing this sort of check just adds overhead for the blocks that do not overlap the diagonal. When only using this optimization with blocks that overlap the diagonal, various NumPy tricks can be applied to speed this up substantially.